src/battle_factory.c

```diff
@@ -741,7 +741,14 @@ u8 GetFactoryMonFixedIV(u8 challengeNum, bool8 isLastBattle)
     u8 ivSet;
     bool8 useHigherIV = isLastBattle ? TRUE : FALSE;
 
+// The Factory has an out-of-bounds access when generating the rental draft for round 9 (challengeNum==8), 
+// or the "elevated" rentals from round 8 (challengeNum+1==8)
+// This happens to land on a number higher than 31, which is interpreted as "random IVs"
+#ifdef BUGFIX
+    if (challengeNum > 7)
+#else
     if (challengeNum > 8)
+#endif
         ivSet = 7;
```

Comment:
We have an `ARRAY_COUNT` macro for things like this, so that it will continue to work if someone changes the size of `sFixedIVTable`:

```c
#ifdef BUGFIX
    if (challengeNum >= ARRAY_COUNT(sFixedIVTable))
#else
    if (challengeNum > ARRAY_COUNT(sFixedIVTable))
#endif
        ivSet = ARRAY_COUNT(sFixedIVTable) - 1;
```